### ISSUE TITLE
linter: [typescript/no-useless-empty-export] false positives in declaration files

### ISSUE THREAD

**Issue by StyleShit** [oxc-project/oxc#25788]
### What version of Oxlint are you using?

latest

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "categories": {
    "correctness": "off"
  },
  "rules": {
    "typescript/no-useless-empty-export": "error"
  }
}
```

### What happened?

The export in this snippet should not be reported in `.d.ts` files. It's not being reported in [typescript-eslint](https://typescript-eslint.io/play/#ts=6.0.2&fileType=.d.ts&code=KYDwDg9gTgLgBAYwgOwM7wIZwLxwIwDcAUEaJLHAN5wC%2BBQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tmUSWWT0AtjUL0AHsQ7R86KImjRpkcGAC%2BINUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3QAacDUjxYAEw7oKUOPFlEAVmSkBfEDqA&tokens=false):

```TS
// file.d.ts
export const a = 1;

export { };
```

[Playground](https://playground.oxc.rs/?options=%7B%22run%22%3A%7B%22lint%22%3Atrue%2C%22formatter%22%3Afalse%2C%22transform%22%3Afalse%2C%22isolatedDeclarations%22%3Afalse%2C%22whitespace%22%3Afalse%2C%22mangle%22%3Afalse%2C%22compress%22%3Afalse%2C%22scope%22%3Atrue%2C%22symbol%22%3Atrue%2C%22cfg%22%3Afalse%7D%2C%22parser%22%3A%7B%22extension%22%3A%22tsx%22%2C%22allowReturnOutsideFunction%22%3Atrue%2C%22preserveParens%22%3Atrue%2C%22allowV8Intrinsics%22%3Atrue%2C%22semanticErrors%22%3Atrue%7D%2C%22linter%22%3A%7B%22config%22%3A%22%7B%5C%22categories%5C%22%3A%7B%5C%22correctness%5C%22%3A%5C%22off%5C%22%7D%2C%5C%22rules%5C%22%3A%7B%5C%22typescript%2Fno-useless-empty-export%5C%22%3A%5C%22error%5C%22%7D%7D%22%7D%2C%22formatter%22%3A%7B%22useTabs%22%3Afalse%2C%22tabWidth%22%3A2%2C%22endOfLine%22%3A%22lf%22%2C%22printWidth%22%3A80%2C%22singleQuote%22%3Afalse%2C%22jsxSingleQuote%22%3Afalse%2C%22quoteProps%22%3A%22as-needed%22%2C%22trailingComma%22%3A%22all%22%2C%22semi%22%3Atrue%2C%22arrowParens%22%3A%22always%22%2C%22bracketSpacing%22%3Atrue%2C%22bracketSameLine%22%3Afalse%2C%22objectWrap%22%3A%22preserve%22%2C%22singleAttributePerLine%22%3Afalse%7D%2C%22transformer%22%3A%7B%22target%22%3A%22es2015%22%2C%22useDefineForClassFields%22%3Atrue%2C%22experimentalDecorators%22%3Atrue%2C%22emitDecoratorMetadata%22%3Atrue%2C%22optimizeEnums%22%3Atrue%2C%22optimizeConstEnums%22%3Atrue%7D%2C%22isolatedDeclarations%22%3A%7B%22stripInternal%22%3Afalse%7D%2C%22codegen%22%3A%7B%22normal%22%3Atrue%2C%22jsdoc%22%3Atrue%2C%22annotation%22%3Atrue%2C%22legal%22%3Atrue%7D%2C%22compress%22%3A%7B%7D%2C%22mangle%22%3A%7B%22topLevel%22%3Atrue%2C%22keepNames%22%3Afalse%7D%2C%22controlFlow%22%3A%7B%22verbose%22%3Afalse%7D%2C%22inject%22%3A%7B%22inject%22%3A%7B%7D%7D%2C%22define%22%3A%7B%22define%22%3A%7B%7D%7D%7D&code=export+const+a+%3D+1%3B%0A%0Aexport+%7B+%7D%3B&t=ast&lintRules=typescript%2Fno-useless-empty-export)
